staging/src/k8s.io/apiserver/pkg/util/flowcontrol/dropped_requests_tracker_test.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	testingclock "k8s.io/utils/clock/testing"
+)
+
+func TestDroppedRequestsTracker(t *testing.T) {
+	fakeClock := testingclock.NewFakeClock(time.Now())
+	tracker := newDroppedRequestsTracker(fakeClock.Now)
+
+	// The following table represents the list over time of:
+	// - seconds elapsed (as computed since the initial time)
+	// - requests that will be recorded as dropped in a current second
```

Comment:
I find the current design of the test cases to be harder to think about than I think is necessary.  The usual design of a test case is that it say "do _this_ then expect _that_" (for some _this_ and _that_).  The current design does not follow that pattern.  I would find it easier to think about these test cases if the secondsElapsed and droppedRequests prescribed dropping at a steady rate (secondsElapsed/droppedRequests between drops, with no drop at the start and a drop at the end); that would make a coherent "do this" part and it would precede the "expect that" part.